app/assets/javascripts/clipboard.js

```diff
@@ -0,0 +1,37 @@
+$(document).on("turbolinks:load", function () {
+  addCopyButton();
+});
+
+function addCopyButton() {
+  new Clipboard('.clipboard-copy');
+
+  $("pre").each(function(index) {
+    var id = "id_" + index;
+    $(this).children("code").attr("id", id);
+    prependButtonAndHide($(this), id);
+    toggleCopyButton($(this));
+  });
+}
+
+function prependButtonAndHide(pre, id) {
+  if (pre.find(".clipboard-copy").length == 0) {
+    var button = $("<div class='ui small basic icon buttons right floated'>" +
```

Comment:
it's been a while but i forget if you actually need to wrap this in jQuery. you might be able to just prepend `button` as a string but not sure.